artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessageObjectPools.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.message.impl;
+
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.utils.collections.TypedProperties;
+
+public class CoreMessageObjectPools {
+
+   private SimpleString.ByteBufSimpleStringPool addressDecoderPool;
+   private TypedProperties.TypedPropertiesDecoderPools propertiesDecoderPools;
+
+   private SimpleString.StringSimpleStringPool addressStringSimpleStringPool;
+   private TypedProperties.TypedPropertiesStringSimpleStringPools propertiesStringSimpleStringPools;
+
+   public CoreMessageObjectPools() {
+   }
+
+   public SimpleString.ByteBufSimpleStringPool getAddressDecoderPool() {
+      if (addressDecoderPool == null) {
```

Comment:
I would use the Guava's `Suppliers.memoize` for any of the lazy (but concurrent) pool instances in order to:
- have a more concise code here
- synchronized guard that scale for each pool instance (they are not sharing the `this` monitor)

If not, is needed to:
- declare volatile each pool instance
- provide different Object guards to be used to synchronize (eg `Object addressStringSimpleStringPoolGuard = new Object()`)

Reply:
This if the pool instances properties need to be accessed concurrently, TBH i'm not sure they are!